spec/controllers/junior_stroy_controller_spec.rb

```diff
@@ -2,22 +2,80 @@
 
 describe JuniorStoriesController do
   describe 'GET index' do
-    it 'assigns @junior_stories' do
-      story = build :junior_story
-      story.save
-      get :index
-      expect(assigns(:junior_stories)).to eq([story])
+    context 'without filter params' do
+      it 'assigns @junior_stories' do
+        story = build :junior_story
+        story.save
+        get :index
+        expect(assigns(:junior_stories)).to eq([story])
+      end
+
+      it 'renders the index template' do
+        get :index
+        expect(response).to render_template('index')
+      end
+
+      it 'does not render index, but instead responds with csv' do
+        get :index, format: 'csv'
+        expect(response).not_to render_template('index')
+        expect(response.content_type).to include 'application/csv'
+      end
     end
 
-    it 'renders the index template' do
-      get :index
-      expect(response).to render_template('index')
+    context 'with legal filter params' do
+      it 'returns only the filter appropriate junior_story' do
+        female_story = build :junior_story
+        female_story.gender = 'female'
+        female_story.save
+
+        male_story = build :junior_story
+        male_story.gender = 'male'
+        male_story.save
+
+
+        get :index, { 'gender' => 'female' }
+        expect(assigns(:junior_stories)).to eq([female_story])
+      end
+
+      it 'renders the index template' do
+        get :index, { 'gender' => 'female' }
+        get :index
+        expect(response).to render_template('index')
+      end
+
+      it 'does not render index, but instead responds with csv' do
+        get :index, { 'gender' => 'female', 'format' => 'csv' }
+        expect(response).not_to render_template('index')
+        expect(response.content_type).to include 'application/csv'
+      end
     end
 
-    it 'does not render index, but instead responds with csv' do
-      get :index, format: 'csv'
-      expect(response).not_to render_template('index')
-      expect(response.content_type).to include 'application/csv'
+    context 'with illegal filter params' do
+      it 'ignores illegal filter params but respects legal filter params' do
```

Comment:
As I can see, you are testing here the possibility of someone sending illegal params through the filter... is that right? Does this need testing? Isn't this handled by strong parameters anyway?